Automattic/wp-calypso
# PostShare: janitoral

This PR does a janitorial to the post-share component:

* rewrite the component using `class`
* use localize helper
* remove ESLint warnings
* move the component to its own folder
* update dependencies
* janitorial

<img src="https://cloud.githubusercontent.com/assets/77539/24382844/9d611f40-131e-11e7-8ddc-5db846f4d759.png" width="500px" />

<img src="https://cloud.githubusercontent.com/assets/77539/24382863/bb5d36e6-131e-11e7-8ca4-a1eadd029131.png" width="400px" />


### Testing

This PR doesn't add any new feature, so ...

1) Go to posts list of a site: `http://calypso.localhost:3000/posts/my/<site>`
2) Open the `Share` tab for some post
3) Test the whole section. Everything should be ok.
